tests/test_queue.py

```diff
@@ -0,0 +1,129 @@
+import os
+import random
+
+import logging
+from pathlib import Path
+from typing import List 
+
+import pytest
+
+from psij import Job, JobSpec, JobAttributes, JobExecutor, ResourceSpecV1
+from tempfile import TemporaryDirectory
+
+from executor_test_params import ExecutorTestParams
+from _test_tools import _get_executor_instance, _get_timeout, assert_completed, _make_test_dir
+
+
+SCHEDULER_COMMANDS = {
+    "slurm": {
+        "get_queues": "mdiag -c",
+        "get_user_jobs": "squeue -u $(whoami)",
+        "kill_command": "scancel"
+    },
+    "lsf": {
+        "get_queues": "bqueues -u $(whoami)",
+        "get_user_jobs": "bjobs",
+        "kill_command": "bkill"
+    }
+}
+
+
+def get_slurm_queues() -> List[str]:
+    out = os.popen("mdiag -c").read().split("\n")
+    return [line.split("=")[-1] for line in out if "PartitionName" in line]
+
+
+def get_lsf_queues() -> List[str]:
+    valid_queues = []
+    out = "".join(os.popen("bqueues -u $(whoami) -o 'QUEUE_NAME NJOBS PEND RUN SUSP STATUS'").read()).split("\n")
+    out = [l for l in out if len(l) != 0]
+    out = [l.split(" ") for l in out]
+    if len(out) == 0:
+        return []
+    out.pop(0) # popping headers 
+
+    for queue_info in out: 
+        name = queue_info[0]
+        njobs = int(queue_info[1])
+        pend = int(queue_info[2])
+        run = int(queue_info[3])
+        susp = int(queue_info[4])
+        status = str(queue_info[5])
+
+        if "active" not in status.lower():
+            continue
+
+        if (njobs + pend + run + susp) > 10:
+            valid_queues.append(name)
+
+    return valid_queues
+
+
+def get_queue_info(executor: str) -> List[str]:
+    res = []
+    command = SCHEDULER_COMMANDS[executor]["get_user_jobs"]
+    res.extend(os.popen(command).read().split("\n"))
+    return res
+
+
+def kill_job(scheduler: str, job: Job) -> None:
+    command = f"{SCHEDULER_COMMANDS[scheduler]['kill_command']} {job._native_id}"
+    print("Kill command:", command)
+    os.system(command)
+
+
+def make_job(queue:str) -> Job:
+    return Job(
+        JobSpec(
+            executable="/bin/date",
+            attributes=JobAttributes(
+                queue_name=queue,
+            ),
+        )
+    )
+
+
+def test_queue(execparams: ExecutorTestParams) -> None:
+    scheduler = ""
+    queues = []
+    slurm_queues = get_slurm_queues()
+    lsf_queues = get_lsf_queues()
+
+    queues.extend(slurm_queues)
+    queues.extend(lsf_queues)
+
+    if len(slurm_queues) != 0:
+        scheduler = "slurm"
+    elif len(lsf_queues) != 0:
+        scheduler = "lsf"
+
+    if len(queues) < 2:
+        pytest.raises(Exception("Need at least two queues to perform this test"))
```

Comment:
I think you are looking for `pytest.skip` instead.
`pytest.raises` is used to test that a block of code throws a specific exception. For example,

```python
def test_that_division_by_zero_correctly_raises_exception():
    with pytest.raises(ZeroDivisionError):
        1 / 0
```

In other words you use it to check that you test throws an exception. If you had `1 / 1` instead of `1 / 0`, `pytest.raises` would actually cause the test to fail because it did not throw the exception that was expected.